hw/ip/rv_dm/data/rv_dm_testplan.hjson

```diff
@@ -463,6 +463,22 @@
       stage: V1
       tests: ["rv_dm_halt_resume_whereto"]
     }
+    {
+      name: progbuf_read_write_execute
+      desc: '''
+            Verify the functionality of program buffer by writing random data on program buffer.
+
+            - Program jtag_dmi_ral.dmcontrol.haltreq to 1'b1.
+            - Acknowledge haltreq by setting tl_mem_ral.halted to 0.
```

Comment:
I know that I've said this on another PR, but I don't think this line quite makes sense. Please fix it to talk about waiting until `halted` becomes the idcode.

Reply:
Sorry, I was confused about this (on the other PR as well). But don't we have to see the halt signal coming out of the debug module before we tell it that we're obeying the signal? I think there's a step missing here.

Reply:
ok i have added a step . i will do a separate PR as in this pr there are conflicts.

Reply:
This looks good, thanks.